cumulusci/tasks/marketing_cloud/deploy.py

```diff
@@ -84,7 +86,7 @@ def _poll_action(self):
         )
         result = safe_json_from_response(response)
         self.logger.info(f"Waiting [{result['status']}]...")
-        if result["status"] == "DONE":
+        if result["status"] in [DEPLOY_FINISHED_STATUS, "FATAL_ERROR"]:
             self.poll_complete = True
             self._validate_response(result)
```

Comment:
I'd suggest something like this so we can keep `_validate_response` focused on handling the case where the deployment reached DONE status:
```suggestion
        if result["status"] == DEPLOY_FINISHED_STATUS:
            self.poll_complete = True
            self._validate_response(result)
        elif result["status"] == "FATAL_ERROR":
            self.poll_complete = True
            self._report_fatal_error(result)
```

And can we make sure that _report_fatal_error logs the entire `result` (I don't know if it will give us any more information, but if we don't log it we won't find out)